### PR TITLE
Wasm2js Atomics support

### PR DESCRIPTION
**Pull request by kripken**
Atomic loads, stores, RMW, cmpXchg, wait, and notify. This is enough
to get the JS atomics tests in the emscripten test suite to pass, at least.

### CODE REVIEW

File: src/wasm2js.h [hunk 1]
```diff
@@ -2446,6 +2541,67 @@ void Wasm2JSGlue::emitSpecialSupport() {
...
+    if (result == 'ok') return 0;
+    if (result == 'not-equal') return 1;
+    if (result == 'timed-out') return 2;
+    throw 'bad result ' + result;
+  }
+      )";
+    } else if (import->base == ABI::wasm2js::ATOMIC_RMW_I64) {
+      out << R"(
+  function wasm2js_atomic_rmw_i64(op, bytes, offset, ptr, valueLow, valueHigh) {
+    assert(bytes == 8); // TODO
```
Comment (tlively) [inline src/wasm2js.h:2561]: What's the TODO here?

File: src/wasm2js.h [hunk 2]
```diff
@@ -2446,6 +2541,67 @@ void Wasm2JSGlue::emitSpecialSupport() {
...
+    var view = new BigInt64Array(bufferView.buffer); // TODO cache
+    ptr = (ptr + offset) >> 3;
+    var value = BigInt(valueLow >>> 0) | (BigInt(valueHigh >>> 0) << BigInt(32));
```
Comment (tlively) [inline src/wasm2js.h:2564]: I guess this assumes the browser has BigInt but no BigInt WebAssembly integration? Does that describe most browsers?

File: src/wasm2js.h (hunk 1 above)
Comment (kripken) [reply src/wasm2js.h:2561]: To support anything else than 8 :wink: I can clarify.

File: src/wasm2js.h (hunk 2 above)
Comment (kripken) [reply src/wasm2js.h:2564]: I think so, yeah, https://caniuse.com/#feat=bigint vs https://webassembly.org/roadmap/

But why do you say this code assumes there is no bigint wasm integration?

Comment (kripken) [reply src/wasm2js.h:2564]: Oh, because it doesn't return a bigint? Yeah, the code assumes wasm MVP.

**Pull request merged**